### PR TITLE
	Fixed and modernised docker / docker-compose setup

### PR DESCRIPTION
**Pull request by rawkode**
- Switched to docker-compose syntax version: '2'
- Updating docker-compose to provide a tmpfs cache directory that won't sync / pollute local disk

### CODE REVIEW

File: Dockerfile [hunk 1]
```diff
@@ -1,12 +1,11 @@
...
- && docker-php-ext-install zip \
- && a2enmod rewrite \
- && sed -i 's!/var/www/html!/var/www/public!g' /etc/apache2/apache2.conf \
- && mv /var/www/html /var/www/public \
- && curl -sS https://getcomposer.org/installer \
-  | php -- --install-dir=/usr/local/bin --filename=composer
+  && apt-get install -y git zlib1g-dev \
+  && docker-php-ext-install zip \
+  && a2enmod rewrite \
+  && sed -i 's!/var/www/html!/var/www/public!g' /etc/apache2/sites-enabled/000-default.conf \
```
Comment (webimpress) [inline Dockerfile]: Please have a look on apigility version of this file:
https://github.com/zfcampus/zf-apigility-skeleton/blob/master/Dockerfile
We have there also:
` && mv /var/www/html /var/www/public \`

Isn't it required here?

Comment (rawkode) [inline Dockerfile]: `docker-compose` mounts your local directory to `/var/www`, removing whatever is there in the base `Dockerfile`; we simply need to do the `sed` command to update the `VirtualHost` and everything is fine, as the `public` directory exists within the mount.

**Comment by adamlundrigan** [comment]
#389 and #390 propose similar changes.  I vote this PR be accepted and the others closed.

**Comment by rawkode** [comment]
@adamlundrigan  Anything I can do to help massage this through?

**Comment by bitwombat** [comment]
The README.md also need to correct:

docker-compose up -d --build
to
docker-compose up -d

**Comment by bitwombat** [comment]
It should be noted that this change requires docker-compose 1.6+.  Ubuntu 16.04 LTS has 1.5.2 currently.

Not sure what minimum versions of docker stuff the team wants to require.  Is docker-compose 2.0 backwards compatible?

The fixes to the sed are actually an error, and should probably merged right away, as in  #390

**Comment by adamlundrigan** [comment]
@rawkode not sure who is the maintainer of this repository currently, but @weierophinney originally added the Docker stuff so I will tag him.

**Comment by rawkode** [comment]
Shall I just close this? @weierophinney @adamlundrigan

**Pull request closed without merging**